### PR TITLE
Unit tests for reducer.

### PR DESCRIPTION
**Pull request by rchen20**
# Summary

- Adds reducer unit tests for feature/test-reorg.

1. Adds constructor, init, and reset tests for CPU and OMP.
2. Done: (Normal behavior) Investigate why loc reductions only reset() to default value and 0.
3. Done: CUDA.  HIP and OMP_TARGET will be in separate PR.
4. Will do in separate PR: Reorganize old reduction tests.

This PR implements 1.

### CODE REVIEW

**Comment by artv3** [comment]
Do you plan to merge this once 2,3,4 are complete? Or will those be done in a different PR?

**Comment by rchen20** [comment]
@artv3 Yes, I'd like to do 2-4 before merging. I should be able to finish 2 and 3 pretty quickly. For 4, all of those tests can probably be put into test/functional/, and then broken apart into smaller chunks, unless there are any other ideas.

**Comment by rchen20** [comment]
@MrBurmark @rhornung67 Added CUDA constructor test, and parameterized both tests. Should be ready to go pending approval. I will separate the old reducer tests into functional/ in another PR.

File: test/unit/reducer/test-reduce-constructors.cpp [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+};
+
+#if defined(RAJA_ENABLE_CUDA)
+template <typename T>
+class ReducerCUDAConstructorTest : public ::testing::Test
+{
+};
+#endif
+
+TYPED_TEST_CASE_P(ReducerBasicConstructorTest);
```
Comment (rhornung67) [inline test/unit/reducer/test-reduce-constructors.cpp]: Please replace 'Test' with 'UnitTest' in all test labels.

Comment (rchen20) [reply test/unit/reducer/test-reduce-constructors.cpp]: Thanks, this should be updated now.

**Comment by rhornung67** [comment]
@rchen20 is this PR (and branch) still needed, or was this work merged in another PR?

**Pull request closed without merging**